misc/python/materialize/mzcompose.py

```diff
@@ -619,18 +618,34 @@ def named(cls, name: str) -> Type["WorkflowStep"]:
             raise errors.UnknownItem("step", name, list(cls._steps))
 
     @classmethod
-    def register(cls, name: str) -> Callable[[Type[T]], Type[T]]:
+    def register(
+        cls, name: str
+    ) -> Callable[[Type["WorkflowStep"]], Type["WorkflowStep"]]:
         if name in cls._steps:
             raise ValueError(f"Double registration of step name: {name}")
 
-        def reg(to_register: Type[T]) -> Type[T]:
+        def reg(to_register: Type["WorkflowStep"]) -> Type["WorkflowStep"]:
             if not issubclass(to_register, WorkflowStep):
                 raise ValueError(
                     f"Registered step must be a WorkflowStep: {to_register}"
                 )
             cls._steps[name] = to_register
             to_register.name = name
-            return to_register  # type: ignore
+
+            # Allow the step to also be called as a Step.step_name() classmethod
+            def run_step(workflow: Workflow, **args: Dict) -> None:
+                step: WorkflowStep = to_register(**args)
+                step.run(workflow)
+
+            func_name = name.replace("-", "_")
+            if not hasattr(cls, func_name):
+                setattr(cls, func_name, run_step)
+            else:
+                raise errors.Failed(
+                    f"Unable to register method Step.{func_name} as one already exists."
+                )
```

Comment:
I think it would be easier to use this (autocompletion would work, mypy wouldn't complain) if we just made `run_step()` a global function in this module and required every step to define themselves as a function:

```python
# unfortunately handling higher order functions with kwargs is rough
# https://stackoverflow.com/a/61583234/25616
def runnable(step: WorkflowStep) -> Callable[[Workflow, Any], None]:
    def run_step(workflow: Workflow, **args: Dict) -> None:
       step: WorkflowStep = step(**args)
       step.run(workflow)
    run_step.name = step.name
    return run_step


@Steps.register("good-name")
class GoodNameStep(WorkflowStep): ...

good_name = runnable(GoodNameStep)
```

I would really like for us to make WorkflowStep classes unnecessary entirely and just create a decorator that can work on functions directly. I haven't gotten around to it yet, but it would be great.

Reply:
can I auto-register the steps as global functions . This will not work with autocompletion, but will not require every new step to be registered separately as a global function in the future. 

I think the Step classes should stay as you can then ask them to pretty-print themselves, dump themselves back to YAML and other such stuff, if ever necessary.

Reply:
They shouldn't be global functions but just methods on the `Workflow` class. That would be most ergonomic, I think. 

Long term I would like to entirely deprecate writing workflows in YAML.